### PR TITLE
[GEOS-9526] review extension license information

### PR DESCRIPTION
**Pull request by jodygarnett**
Working on a couple of glitches:

* [x] maven assembly file inclusion sensitive to location of `mvn -f src/pom.xml`
* [x] upgrade to maven assembly plugin 3.2.0 to use of `mvn assembly:single`
* [x] review extension licenses
* [x] Avoid multiple files called LICENSE.txt by changing to GPL.md and LGPL.md as needed
* [x] use markdown licenses where available
* [x] distinguish between license NOTICE.md (describing how licenses are used) and the licenses themselves

See [GEOS-9526](https://osgeo-org.atlassian.net/browse/GEOS-9526) for detailed discussion.

## Checklist

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira [GEOS-9526](https://osgeo-org.atlassian.net/browse/GEOS-9526)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by jodygarnett** [comment]
looks like I need to update travis doc build command.

File: build/build_release.sh [hunk 1]
```diff
@@ -294,8 +294,7 @@ else
    echo "Skipping mvn clean install -P communityRelease -DskipTests"
 fi
 
-
-mvn assembly:attached $MAVEN_FLAGS
+mvn -f src/pom.xml assembly:single $MAVEN_FLAGS -N
```
Comment (tbarsballe) [inline build/build_release.sh]: The `pushd src` a few lines above means you don't need `-f src/pom.xml` (more specifically, as written the build will fail - remove `-f src/pom.xml` )

**Comment by jodygarnett** [comment]
windows failure unrelated

File: src/release/extensions/css/SCALA-LICENSE.txt [hunk 2]
```diff
@@ -1,35 +0,0 @@
-SCALA LICENSE
```
Comment (tbarsballe) [inline src/release/extensions/css/SCALA-LICENSE.txt]: I see this was deleted, but didn't see a replacement. Any reason why there was no replacement?

Comment (jodygarnett) [reply src/release/extensions/css/SCALA-LICENSE.txt]: This extension is no longer based on scala- Andrea rewrote it some years ago to be pure java.

**Pull request merged**